src/ExpressionEvaluator/Core/Source/ResultProvider/ResultProvider.cs

```diff
@@ -560,23 +597,24 @@ private void GetItemsAndContinue(EvalResultDataItem dataItem, DkmWorkList workLi
             var numRows = rows.Count;
             Debug.Assert(count >= numRows);
             var results = new DkmEvaluationResult[numRows];
-            GetEvaluationResultsAndContinue(rows, results, 0, numRows, workList, inspectionContext, value.StackFrame,
-                () => ContinueWithExceptionHandling(
+            var wl = new WorkList(workList, e => completionRoutine(DkmEvaluationEnumAsyncResult.CreateErrorResult(e)));
+            GetEvaluationResultsAndContinue(rows, results, 0, numRows, wl, inspectionContext, value.StackFrame,
+                () => wl.ContinueWith(
```

Comment:
It looks like we could call wl.ContinueWith outside the lambda (e.g. immediately after construction).